### PR TITLE
refactor(aws): resources migration batch1

### PR DESCRIPTION
**Pull request by sinabakh**
giphy

### CODE REVIEW

**Comment by sinabakh** [comment]
@aliscott I migrated a few resources and fixed and changed all `_gb` keys to float. There seems to be a problem in syncing the usage file that is happening just for `cloudwatch_log_group` that keeps the `.0` for floats. For other resources, it works just fine. I think it might be related to wildcard and module syncing but I couldn't find the problem.

It would be great if you could take a look at the failing test.

**Comment by aliscott** [comment]
@sinabakh I tested out the failing test. I think we just want to update the golden file for this case. The sync usage adds .0 for float values, and I think that makes sense. We already have it for the `monthly_data_processed_gb` in that golden file.

File: internal/providers/terraform/aws/cloudwatch_log_group.go [hunk 1]
```diff
@@ -1,84 +1,18 @@
 package aws
 
 import (
+	"github.com/infracost/infracost/internal/resources/aws"
 	"github.com/infracost/infracost/internal/schema"
-	"github.com/shopspring/decimal"
 )
 
 func GetCloudwatchLogGroupItem() *schema.RegistryItem {
```
Comment (aliscott) [inline internal/providers/terraform/aws/cloudwatch_log_group.go:8]: Might not be possible, but can we update these to be internal functions (`getCloudwatchLogGroupItem`). We've started using that for new resources.

**Comment by sinabakh** [comment]
> @sinabakh I tested out the failing test. I think we just want to update the golden file for this case. The sync usage adds .0 for float values, and I think that makes sense. We already have it for the `monthly_data_processed_gb` in that golden file.

I think that might not be the case, there are other resources that have float usage data with no `.0` and tests are not failing. `transfer_server` is an example, it has `monthly_data_downloaded_gb` and `monthly_data_uploaded_gb` and they are both floats but there is no problem in thinking the usage file for it.

File: internal/providers/terraform/aws/cloudwatch_log_group.go (hunk 1 above)
Comment (sinabakh) [reply internal/providers/terraform/aws/cloudwatch_log_group.go:8]: I think it's easy, I'll applyt it.

**Comment by aliscott** [comment]
> > @sinabakh I tested out the failing test. I think we just want to update the golden file for this case. The sync usage adds .0 for float values, and I think that makes sense. We already have it for the `monthly_data_processed_gb` in that golden file.
> 
> I think that might not be the case, there are other resources that have float usage data with no `.0` and tests are not failing. `transfer_server` is an example, it has `monthly_data_downloaded_gb` and `monthly_data_uploaded_gb` and they are both floats but there is no problem in thinking the usage file for it.

I only see usage data for `aws_transfer_server` that is read by tests, not any that is written/synced by tests (https://github.com/sinabakh/infracost/blob/refactor-migration-aws-batch1/internal/providers/terraform/aws/testdata/transfer_server_test/transfer_server_test.usage.yml).

File: internal/providers/terraform/aws/cloudwatch_log_group.go [hunk 2]
```diff
@@ -1,84 +1,18 @@
...
 	return &schema.RegistryItem{
 		Name:  "aws_cloudwatch_log_group",
-		RFunc: NewCloudwatchLogGroup,
+		RFunc: NewCloudwatchLogGroupItem,
```
[the remark this answers is not in the document]
Comment (sinabakh) [reply internal/providers/terraform/aws/cloudwatch_log_group.go]: It was getting the name from registry func name, fixed it by using the resource actual name.

**Comment by sinabakh** [comment]
@aliscott Ready for review and merge

**Review by aliscott — APPROVED** [review]
Thanks @sinabakh! Will merge this after the v0.9.16 release so it gets into v0.9.17.

**Pull request merged**